examples/AdvancedOperations/AddDynamicPageFeed.php

```diff
@@ -0,0 +1,485 @@
+        self::addDsaTarget($googleAdsClient, $customerId, $adGroupId, $dsaPageUrlLabel);
+
+        printf("Dynamic page feed setup is complete for campaign ID %s.%s", campaignId, PHP_EOL);
+    }
+
+    /**
+    * Creates a feed.
+    *
+    * @param GoogleAdsClient googleAdsClient the Google Ads API client.
+    * @param int customerId the client customer ID in which to create the feed.
+    * @return array an array containing names and IDs of feed attributes.
+    */
+    private static function createFeed(
+        GoogleAdsClient $googleAdsClient,
+        int $customerId
+    ) {
+        // Creates a URL attribute.
+        $urlAttribute = new FeedAttribute([
+            'type' => FeedAttributeType::URL_LIST,
+            'name' => new StringValue(['value' => "Page URL"])
+        ]);
+
+        // Creates a label attribute.
+        $labelAttribute = new FeedAttribute([
+            'type' => FeedAttributeType::STRING_LIST,
+            'name' => new StringValue(['value' => "Label"])
+        ]);
+
+        // Creates the feed.
+        $feed = new Feed([
+            'name' => new StringValue(['value' => 'DSA Feed #' . uniqid()]),
+            'attributes' => [$urlAttribute, $labelAttribute],
+            'origin' => FeedOrigin::USER
+        ]);
+
+        // Creates a feed operation for creating a feed.
+        $feedOperation = new FeedOperation();
+        $feedOperation->setCreate($feed);
+
+        // Issues a mutate request to add the feed.
+        $feedServiceClient = $googleAdsClient->getFeedServiceClient();
+        $feedResponse = $feedServiceClient->mutateFeeds($customerId, [$feedOperation]);
+
+        $feedResourceName = $feedResponse->getResults()[0]->getResourceName();
+        printf("Added feed named '%s'.%s", $feedResourceName, PHP_EOL);
+
+        return self::getFeed($googleAdsClient, $customerId, $feedResourceName);
+    }
+
+    /**
+     * Retrieves details about a feed.
+     *
+     * @param GoogleAdsClient googleAdsClient the Google Ads API client.
+     * @param int customerId the client customer ID.
+     * @param string feedResourceName the resource name of the feed.
+     * @return array an array containing names and IDs of feed attributes.
+     */
+    private static function getFeed(
+        GoogleAdsClient $googleAdsClient,
+        int $customerId,
+        string $feedResourceName
+    ) {
+        $query = "SELECT feed.attributes FROM feed WHERE feed.resource_name = '$feedResourceName'";
+
+        $googleAdsServiceClient = $googleAdsClient->getGoogleAdsServiceClient();
+        $response =
+            $googleAdsServiceClient->search($customerId, $query, ['pageSize' => self::PAGE_SIZE]);
+
+        $feedAttributes = $response->getIterator()->current()->getFeed()->getAttributes();
+        $feedDetails = ["resource_name" => $feedResourceName];
+        foreach ($feedAttributes as $feedAttribute) {
+            $feedDetails[$feedAttribute->getNameValue()] = $feedAttribute->getIdValue();
```

Comment:
Might be great to add a comment in all foreach loops of the file to give the types of the objects (here 'feedAttribute') being used

Reply:
Done.